examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1502 @@
+        system_matrix.reinit(dsp);
+
+        // Assemble system matrix by calling MatrixFreeTools::compute_matrix().
+        MatrixFreeTools::compute_matrix(
+          matrix_free,
+          constraints,
+          system_matrix,
+          &LaplaceOperator::do_cell_integral_local,
+          this);
+      }
+
+    return this->system_matrix;
+  }
+
+
+
+  template <int dim, typename number>
+  void LaplaceOperator<dim, number>::do_cell_integral_local(
+    FECellIntegrator &integrator) const
+  {
+    integrator.evaluate(EvaluationFlags::gradients);
+
+    for (unsigned int q = 0; q < integrator.n_q_points; ++q)
+      integrator.submit_gradient(integrator.get_gradient(q), q);
+
+    integrator.integrate(EvaluationFlags::gradients);
+  }
+
+
+
+  template <int dim, typename number>
+  void LaplaceOperator<dim, number>::do_cell_integral_global(
+    FECellIntegrator &integrator,
+    VectorType &      dst,
+    const VectorType &src) const
+  {
+    integrator.gather_evaluate(src, EvaluationFlags::gradients);
+
+    for (unsigned int q = 0; q < integrator.n_q_points; ++q)
+      integrator.submit_gradient(integrator.get_gradient(q), q);
+
+    integrator.integrate_scatter(EvaluationFlags::gradients, dst);
+  }
+
+
+
+  template <int dim, typename number>
+  void LaplaceOperator<dim, number>::do_cell_integral_range(
+    const MatrixFree<dim, number> &              matrix_free,
+    VectorType &                                 dst,
+    const VectorType &                           src,
+    const std::pair<unsigned int, unsigned int> &range) const
+  {
+    FECellIntegrator integrator(matrix_free, range);
+
+    for (unsigned cell = range.first; cell < range.second; ++cell)
+      {
+        integrator.reinit(cell);
+
+        do_cell_integral_global(integrator, dst, src);
+      }
+  }
+
+
+
+  // @sect3{Solver and preconditioner}
+
+  // The following parameter set controls the geometric multigrid mechanism and
+  // the solver specifications on the coarsest level. We populate it with
+  // default parameters.
+  struct GMGParameters
+  {
+    struct CoarseSolverParameters
+    {
+      std::string  type            = "cg_with_amg";
+      unsigned int maxiter         = 10000;
+      double       abstol          = 1e-20;
+      double       reltol          = 1e-4;
+      unsigned int smoother_sweeps = 1;
+      unsigned int n_cycles        = 1;
+      std::string  smoother_type   = "ILU";
+    };
+
+    struct SmootherParameters
+    {
+      std::string  type                = "chebyshev";
+      double       smoothing_range     = 20;
+      unsigned int degree              = 5;
+      unsigned int eig_cg_n_iterations = 20;
+    };
+
+    SmootherParameters     smoother;
+    CoarseSolverParameters coarse_solver;
+
+    MGTransferGlobalCoarseningTools::PolynomialCoarseningSequenceType
+      p_sequence = MGTransferGlobalCoarseningTools::
+        PolynomialCoarseningSequenceType::decrease_by_one;
+    bool perform_h_transfer = true;
+  };
```

Comment:
I would suggest to integrate this into the other parameter struct. Generally, I would suggest to move the parameters to the top of the file.

Reply:
I agree on the integration part.

I don't feel strongly on where to put it in the file.